crates/metal-mir/src/parcel.rs

```diff
@@ -5,37 +5,37 @@
 use crate::{stmt, types};
 
 #[derive(Debug, Clone)]
-pub struct Parcel {
+pub struct Parcel<'a> {
     /// Parcel name.
     /// i.e. `std`
     pub name: String,
     /// Modules included in
     /// this parcel.
-    pub modules: Vec<Module>,
+    pub modules: Vec<Module<'a>>,
 }
 
 #[derive(Debug, Clone)]
-pub struct Module {
+pub struct Module<'a> {
     /// Module name.
     /// i.e. `io`
     pub name: String,
     /// If this is a submodule
     /// includes the higher-level parent.
-    pub parent: Option<Box<Module>>,
+    pub parent: Option<Box<&'a Module<'a>>>,
```

Comment:
isn't box unnecessary now? same for all the other cases

Reply:
> isn't box unnecessary now? same for all the other cases

This is mentioning the Module struct itself though. As far as I'm aware, you can't reference the same struct in a struct without a box, could be wrong though.

Reply:
you can as long as there's an indirection, afaik &/& mut should work here fine

Reply:
Fixed